engine/src/main/java/org/terasology/recording/EventSystemReplayImpl.java

```diff
@@ -0,0 +1,728 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.recording;
+
+import org.terasology.audio.events.PlaySoundEvent;
+import org.terasology.engine.paths.PathManager;
+import org.terasology.entitySystem.entity.internal.EngineEntityManager;
+import org.terasology.entitySystem.event.internal.EventReceiver;
+import org.terasology.entitySystem.event.internal.EventSystem;
+import com.esotericsoftware.reflectasm.MethodAccess;
+import com.google.common.base.Objects;
+import com.google.common.base.Predicates;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Queues;
+import com.google.common.collect.SetMultimap;
+import com.google.common.collect.Sets;
+import org.reflections.ReflectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.engine.SimpleUri;
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.AbstractConsumableEvent;
+import org.terasology.entitySystem.event.ConsumableEvent;
+import org.terasology.entitySystem.event.Event;
+import org.terasology.entitySystem.event.EventPriority;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.event.internal.PendingEvent;
+import org.terasology.entitySystem.metadata.EventLibrary;
+import org.terasology.entitySystem.metadata.EventMetadata;
+import org.terasology.entitySystem.systems.ComponentSystem;
+import org.terasology.input.BindAxisEvent;
+import org.terasology.input.BindableButton;
+import org.terasology.input.events.KeyEvent;
+import org.terasology.input.events.MouseButtonEvent;
+import org.terasology.input.events.MouseWheelEvent;
+import org.terasology.logic.characters.CharacterMoveInputEvent;
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.network.BroadcastEvent;
+import org.terasology.network.Client;
+import org.terasology.network.NetworkComponent;
+import org.terasology.network.NetworkEvent;
+import org.terasology.network.NetworkMode;
+import org.terasology.network.NetworkSystem;
+import org.terasology.network.OwnerEvent;
+import org.terasology.network.ServerEvent;
+import org.terasology.world.block.BlockComponent;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.*;
+import java.util.concurrent.BlockingQueue;
+
+
+
+/**
+ * Event System used during a replay. It works almost the same as EventSystemImpl, but it considers the recorded events
```

Comment:
If this is the central most class in this feature could you extend on the javadoc a bit please?